crates/typst-syntax/src/parser.rs

```diff
@@ -157,7 +155,14 @@ fn heading(p: &mut Parser) {
 
 /// Parses an item in a bullet list: `- ...`.
 fn list_item(p: &mut Parser) {
-    p.with_nl_mode(AtNewline::RequireColumn(p.current_column()), |p| {
+    let mut column = 0;
+    for i in (0..p.current_start()).rev() {
+        if p.text.chars().nth(i) == Some('\n') {
+            column = i;
+            break;
+        }
+    }
+    p.with_nl_mode(AtNewline::RequireColumn(p.current_start() - column), |p| {
         let m = p.marker();
         p.assert(SyntaxKind::ListMarker);
         markup(p, false, false, syntax_set!(RightBracket, End));
```

Comment:
```suggestion
        markup(p, true, false, syntax_set!(RightBracket, End));
```

Instead we should just change the `at_start` boolean to `true` on the nested call to `markup()` here. This should also happen for enums, and for the _second_ call to `markup()` for term items if we want to be consistent.